### DISCUSSION TITLE
Bundle the generated code into an artifact

### DISCUSSION THREAD

**Discussion by FreifeldRoyi** [quarkiverse/quarkus-openapi-generator#899] [Q&A]
Hi all!
I was wandering if anyone tried to pack the generated classes as an artifact and upload to a artifact management like nexus/artifactory/etc...

I would like to create a package from each of my services and upload it as a standalone package.

My thoughts are to create rather generic `pom.xml` to add to the generated code and in the CI pipeline, copy it to the right place and bundle everything together.
Do you have a better solution?

**Comment by ricardozanini** [comment]
You can create your project with just `src/main/openapi` and bundle all the specs you want for the library there. Then, once compiled, the generated classes will be made available in the classpath. Just ship it.

**Reply by FreifeldRoyi** [reply]
Thank you @ricardozanini for the reply.
The process is a bit different:
I first write the code with REST endpoint and implementation of service A. From the code, I generate the `openapi` file and than the clients of service A.
I would like to ship ONLY the clients as a rather small dependency for the other services that are going to use service A, and not all the service's code. It makes much more sense to me, unless you think i'm missing something. 
The code is generated in the `target/generated-sources/open-api-json` folder. It doesn't have any maven or gradle files of its own to ship it. The only maven in that context is of service A.

**Reply by ricardozanini** [reply]
@FreifeldRoyi I'm not following. You can create a simple project with this generated openapi file in `src/main/openapi`. The code in the `target/generated-sources/open-api-json` will be available in the `.jar` file after compilation. You then distribute this file.

If you really want to add it to `src/main/java`, then you can use Maven Assembly plugin. Also, I think with the right setup you can easily do it from your service (which generated the openapi file) to use maven assembly plugin to generate the right jar file for you.

**Reply by FreifeldRoyi** [reply]
Thnx. I'll take a look into assembly plugin again.
As far as I recall, it just bundles whatever in the target folder with a few simple changes. I don't remember that I can use it to create my own custom jar file
